core/client/app/models/post.js

```diff
@@ -63,6 +64,21 @@ export default Model.extend(ValidationEngine, {
     scratch: null,
     titleScratch: null,
 
+    // Computed Date properties (adding blog settings timezone offset to utc time)
+
+    publishedAtOffset: computed('timeZone.offset', 'publishedAt', function() {
+        let publishedAt = this.get('publishedAt');
```

Comment:
When there is a fair bit of duplication in an if/else conditional like the one below it can usually be eliminated by identifying the core differences and extracting them out into a conditional assignments. For example, in this case it's possible to simplify to something like this:

```javascript
let publishedAt = this.get('publishedAt');
let momentPublishedAt = publishedAt ? moment(publishedAt) : moment.utc();

return this.get('timeZone.offset').then((offset) => {
    return momentPublishedAt.utcOffset(offset);
});
```

There's now less chance of accidental mistakes where one code path diverges from the other and it can be easier to read as the key differentiation is up-front rather than hidden inside the conditional branches.